### ISSUE TITLE
Support for websocket

### ISSUE THREAD

**Issue by kuzmi4** [umputun/reproxy#79]
Need websocket support

**Comment by umputun** [comment]
can you provide more detail on what exactly is missing now? The actual proxy used by reporxy is stdlib's `httputil.ReverseProxy` and it supports WS for many years (since v1.2)

As far as I recall @bobuk actually used reproxy with SW and may provide some inside

**Comment by kuzmi4** [comment]
I use reproxy for public hasura api (graphql server). Hasura user websocket for subscrubtion query.


Message in console when i make request to hasura endpoint:
```
reproxy           | 2021/05/22 15:21:06.701 [WARN]  {lgr/adaptor.go:16 lgr.(*Writer).Write} http: proxy error: can't switch protocols using non-Hijacker ResponseWriter type struct { http.ResponseWriter }
reproxy           | 2021/05/22 15:22:09.156 [DEBUG] {proxy/proxy.go:226 proxy.(*Http).proxyHandler.func2} proxy to http://172.18.0.4:8080/v1/graphql
reproxy           | 2021/05/22 15:22:09.163 [WARN]  {lgr/adaptor.go:16 lgr.(*Writer).Write} http: proxy error: can't switch protocols using non-Hijacker ResponseWriter type struct { http.ResponseWriter }
reproxy           | 2021/05/22 15:23:09.300 [DEBUG] {proxy/proxy.go:226 proxy.(*Http).proxyHandler.func2} proxy to http://172.18.0.4:8080/v1/graphql

 ```


My docker-compose file:

```
version: "3"
services:
  reproxy:
    image: umputun/reproxy
    container_name: reproxy
    hostname: reproxy
    ports:
      - "80:8080"
      - "443:8443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - DEBUG=true
      - TZ=Europe/Moscow
      - DOCKER_ENABLED=true
      - MAX_SIZE=0
      - GZIP=true

......


  graphql-engine:
    image: hasura/graphql-engine:v2.0.0-alpha.10
    depends_on:
      - "db"
    restart: always
    ports:
     - 127.0.0.1:8080:8080
    env_file:
      - .env_main
    labels:
      - "reproxy.route=^/api/hasura/(.*)"


```

**Comment by umputun** [comment]
thx for the details. Looks like some of the handlers redefine ResponseWriter without implementing Hijacker interface.

I have two suspects - gzip handler or the metrics handler. Most likely this is due to metrics, which are partially enabled (but not exposed) by default. I have changed the default to eliminate them completely. The master image built and you can test it by changing `image: umputun/reproxy` to `image: umputun/reproxy:master`

Let me know if helped

**Comment by kuzmi4** [comment]
thx, master version is working.


P.S>I checked before with gzip=off/on and it,s no helped.

**Comment by umputun** [comment]
cool, I'll add a fix to metrics middleware

**Comment by umputun** [comment]
FYI: just released a hotfix [v0.6.2](https://github.com/umputun/reproxy/releases/tag/v0.6.2)
